src/Enzo/enzo_EnzoMethodGrackle.cpp

```diff
@@ -29,77 +29,54 @@ EnzoMethodGrackle::EnzoMethodGrackle
   // Method function that can be called to obtain a list of
   // needed field
 
+  // special container for ensuring colour fields are properly grouped
+  const int rank = cello::rank();
   std::vector<std::string> colour_fields;
 
-  if (grackle_data->metal_cooling > 0){
-    std::string metal_name = "metal_density";
 
-    if (! (field_descr->is_field(metal_name))){
-      this->required_fields_.push_back(metal_name);
-    }
-    colour_fields.push_back(metal_name);
+  this->required_fields_ = std::vector<std::string> {"density","internal_energy",
+                                                     "total_energy"};
+  if (rank>=0) this->required_fields_.push_back("velocity_x");
```

Comment:
Off-by-one error--should be rank >= 1,2,3.  I think this occurs in several places.

Reply:
Fixed!